tests/interfaces/test_batch_autograd_qnode.py

```diff
@@ -989,32 +1009,39 @@ def circuit(a, b):
 
         jac_fn = qml.jacobian(circuit)
         g = jac_fn(a, b)
+        assert isinstance(g, tuple) and len(g) == 2
 
-        expected_g = [
-            [-0.5 * np.sin(a) * np.cos(b), -0.5 * np.cos(a) * np.sin(b)],
-            [0.5 * np.sin(a) * np.cos(b), 0.5 * np.cos(a) * np.sin(b)],
-        ]
-        assert np.allclose(g, expected_g, atol=tol, rtol=0)
+        expected_g = (
+            [-0.5 * np.sin(a) * np.cos(b), 0.5 * np.sin(a) * np.cos(b)],
+            [-0.5 * np.cos(a) * np.sin(b), 0.5 * np.cos(a) * np.sin(b)],
+        )
+        assert np.allclose(g[0], expected_g[0], atol=tol, rtol=0)
+        assert np.allclose(g[1], expected_g[1], atol=tol, rtol=0)
 
         spy = mocker.spy(qml.gradients.param_shift, "transform_fn")
-        hess = qml.jacobian(jac_fn)(a, b)
+        jac_fn_a = lambda *args: jac_fn(*args)[0]
+        jac_fn_b = lambda *args: jac_fn(*args)[1]
+        hess_a = qml.jacobian(jac_fn_a)(a, b)
+        hess_b = qml.jacobian(jac_fn_b)(a, b)
+        assert isinstance(hess_a, tuple) and len(hess_a) == 2
+        assert isinstance(hess_b, tuple) and len(hess_b) == 2
 
         if diff_method == "backprop":
             spy.assert_not_called()
         elif diff_method == "parameter-shift":
             spy.assert_called()
 
-        expected_hess = [
-            [
-                [-0.5 * np.cos(a) * np.cos(b), 0.5 * np.sin(a) * np.sin(b)],
-                [0.5 * np.cos(a) * np.cos(b), -0.5 * np.sin(a) * np.sin(b)],
-            ],
-            [
-                [0.5 * np.sin(a) * np.sin(b), -0.5 * np.cos(a) * np.cos(b)],
-                [-0.5 * np.sin(a) * np.sin(b), 0.5 * np.cos(a) * np.cos(b)],
-            ],
-        ]
-        assert np.allclose(hess, expected_hess, atol=tol, rtol=0)
+        exp_hess_a = (
+            [-0.5 * np.cos(a) * np.cos(b), 0.5 * np.cos(a) * np.cos(b)],
+            [0.5 * np.sin(a) * np.sin(b), -0.5 * np.sin(a) * np.sin(b)],
+        )
+        exp_hess_b = (
+            [0.5 * np.sin(a) * np.sin(b), -0.5 * np.sin(a) * np.sin(b)],
+            [-0.5 * np.cos(a) * np.cos(b), 0.5 * np.cos(a) * np.cos(b)],
+        )
+        for hess, exp_hess in zip([hess_a, hess_b], [exp_hess_a, exp_hess_b]):
+            assert np.allclose(hess[0], exp_hess[0], atol=tol, rtol=0)
+            assert np.allclose(hess[1], exp_hess[1], atol=tol, rtol=0)

```

Comment:
Does this now agree with how JAX returns the Hessian?

Reply:
Kind of. It behaves like
```python
tuple(jax.hessian(fn, argnums=i)(*args) for i in range(len(args)))
```
whereas `jax.hessian(fn, argnums=list(range(len(args))))` also computes off-diagonal blocks (as does Torch, not sure about TF), leading to a two-level tuple (i.e. a 2D array implemented by a tuple of tuples).

Reply:
Is there anything to do/change here, @josh146 ? :)

Reply:
nope, this comment was just for my understanding :)